R/tags.R

```diff
@@ -928,6 +971,17 @@ as.tags.shiny.tag.list <- function(x, ...) {
   x
 }
 
+#' @export
+as.tags.shiny.tag.function <- function(x, ...) {
+  y <- x()
+  # as.tags() doesn't currently have a method for a list of dependencies
+  if (is.list(y) && all(vapply(y, inherits, logical(1), "html_dependency"))) {
+    lapply(y, function(z) as.tags(z))
```

Comment:
Is the `function(z) as.tags(z)` necessary because of S3 dispatch issues?

Reply:
Yep

Reply:
Do you mind adding a comment above, so that we'll remember this in the future? Otherwise someone may try to simplify it and break things.